### DISCUSSION TITLE
Failed to resolve WinRT.Runtime.dll?

### DISCUSSION THREAD

**Discussion by imba-tjd** [microsoft/CsWinRT#1293] [Q&A]
Firstly I didn't install WinSDK or UWP workload.

I `dotnet add package Microsoft.Windows.CsWinRT`, and set TargetFramework=net7.0-windows10.0.22000.0 and CsWinRTMessageImportance=high

dotnet build says:

```
MSBuild version 17.5.0-preview-23061-01+040e2a90e for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  "C:\Users\xxx\.nuget\packages\microsoft.windows.cswinrt\2.0.1\build\tools\IIDOptimizer\IIDOptimizer.exe" @"C:\dotnet\obj\Debug\n
  et7.0-windows10.0.22000.0\IIDOptimizer\cswinrt_iidoptimizer.rsp"
  Failed to resolve WinRT.Runtime.dll.
  dotnet -> C:\dotnet\bin\Debug\net7.0-windows10.0.22000.0\dotnet.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

Is this expected? I googled it but get nothing. Thanks

**Comment by manodasanW** [comment]
In your scenario where you are seeing this, are you generating a C#/WinRT projection?  We have recently seen this where the project has a CsWinRT package reference, but it isn't generating any C#/WinRT projection so the DLL being built has no reference to `WinRT.Runtime`.  This causes the IID Optimizer tool that this error is coming from to fail to find `WinRT.Runtime`.  We do need to improve our message such that it doesn't output in these cases, but the optimizer will just silently not continue, and everything should still build fine.

**Reply by imba-tjd** [reply]
Thanks. I just do some test as a beginner, not for real work.
I have another issue. How do I consume Windows.Win32.winmd? In .NET Framework, I can use `<Reference HintPath>`.
I know CsWin32 but I want to use api via the namespaces defined in winmd, rather than using NativeMethods.txt and Windows.Win32.PInvoke.XXX().

I tried
```xml
  <PropertyGroup>
    <CsWinRTIncludes>Windows.Win32</CsWinRTIncludes>
    <CsWinRTWindowsMetadata>local</CsWinRTWindowsMetadata>
    <CsWinRTMessageImportance>high</CsWinRTMessageImportance>
  </PropertyGroup>

  <ItemGroup>
    <ReferencePath Include="Windows.Win32.winmd" />
    <PackageReference Include="Microsoft.Windows.CsWinRT" Version="2.0.1" />
  </ItemGroup>
```
But it doesn't work.
